testoutputs = [
  '20',
  'output: 0',
  '''output: 1234 has been converted
hello
result: -5
this is a string
this is also a string #### : helloworld
hello''',
'''test1: True
test2: True''',
'''7''',
'''4
6
-1
6
2
True
False
True
False
True
True
True
False
True
True
False''',
'''Outer: 0
Inner: 3
Inner: 2
Inner: 1
Outer end: 0
Outer: 1
Inner: 3
Inner: 2
Inner: 1
Outer end: 1
x is now: 0''',
'120',
'''number is odd
number is even
number is even
number is even
number is even''',
'''aaaaaaaaaaaaaaaa
################################################################################################################################''',
'''1
1
2
3
5
8
13
21
34
55
89
144
Does not exist
420420testTrueTrue'True'
420
should run!
21, odd!
23, odd!
26, even!
30, even!
35, odd!
42, even!
52, even!
70, even!
105, odd!
210, even!''',
'''Enter a number:
2432902008176640000
you did not enter 3
you didn't enter 12 either''',
'''n: 10
m: 10''',
'''Outer: 0
Inner: 3
Inner: 2''',
'1',
'This code should not error',
"hello ' world",
'''5
True
hello world''',
'ErrorType.NAME_ERROR 2',
'''55
5-5''',
'''this should print1
this should print2''',
'''-12345''',
'''this should fail halfway through
ErrorType.SYNTAX_ERROR 3''',
'''i: 10
this is just to make sure that weirdly indented code doesn't break
wonky indent''',
'''foo
bar
defining a func name as '1341545&_+_+*&*^%(*)' should work since it's never called
having syntax errors in an uncalled function should not cause errors
ErrorType.NAME_ERROR 19''',
'''tests wide spacing between tokens
456''',
]